examples/step-61/step-61.cc

```diff
@@ -289,138 +315,178 @@ namespace Step61
+  // exactly the kind of information and operation provided by the
+  // DoFHandler class.
+  //
+  // On the other hand, we don't have such a DoFHandler object for the
+  // Raviart-Thomas space in this program. In fact, we don't even have
+  // an element that can represent the "broken" Raviart-Thomas space
+  // we really want to use here (i.e., the restriction of the
+  // Raviart-Thomas shape functions to individual cells, without the
+  // need for any kind of continuity across cell interfaces. We solve
+  // this conundrum by using the fact that one can call
+  // FEValues::reinit() also with cell iterators into Triangulation
+  // objects (rather than DoFHandler objects). In this case, FEValues
+  // can of course only provide us with information that only
+  // references information of cells, rather than degrees of freedom
+  // enumerated on these cells. So we can't use
+  // FEValuesBase::get_function_values(), but we can use
+  // FEValues::shape_value() to obtain the values of shape functions
+  // at quadrature points on the current cell. It is this kind of
+  // functionality we will make use of below.
+  //
+  // Given this introduction, the following declarations should be
+  // pretty obvious:
   template <int dim>
   void WGDarcyEquation<dim>::assemble_system()
   {
     const FE_RaviartThomas<dim> fe_rt(0);
-    const QGauss<dim>           quadrature_formula(fe_rt.degree + 1);
-    const QGauss<dim - 1>       face_quadrature_formula(fe_rt.degree + 1);
-    const RightHandSide<dim>    right_hand_side;
-
-    // We define objects to evaluate values and
-    // gradients of shape functions at the quadrature points.
-    // Since we need shape functions and normal vectors on faces, we need
-    // <code>FEFaceValues</code>.
-    FEValues<dim> fe_values_rt(fe_rt,
-                               quadrature_formula,
-                               update_values | update_gradients |
-                                 update_quadrature_points | update_JxW_values);
 
-    FEValues<dim> fe_values(fe,
+    const QGauss<dim>     quadrature_formula(fe_rt.degree + 1);
+    const QGauss<dim - 1> face_quadrature_formula(fe_rt.degree + 1);
+
+    FEValues<dim>     fe_values(fe,
                             quadrature_formula,
                             update_values | update_quadrature_points |
                               update_JxW_values);
-
     FEFaceValues<dim> fe_face_values(fe,
                                      face_quadrature_formula,
                                      update_values | update_normal_vectors |
                                        update_quadrature_points |
                                        update_JxW_values);
 
+    FEValues<dim>     fe_values_rt(fe_rt,
+                               quadrature_formula,
+                               update_values | update_gradients |
+                                 update_quadrature_points | update_JxW_values);
```

Comment:
Is this the indentation created by clang-format? It looks a bit strange, but I assume this is due to the 80 character column margin then.

Reply:
Who can tell with clang-format :-) But yes, that's how the tool indents this...